practice/earthquake/site.js

```diff
@@ -0,0 +1,170 @@
+'use strict';
+/**
+ * Makes linter ignore ol variable
+ */
+if (!window) {
+  var ol = '';
+}
+
+var earth = new ol.layer.Tile({
+  source: new ol.source.OSM()
+});
+
+
+
+/**
+ * Requests the earthquake information for the last seven days from the
+ * earthquake USGS database. It returns this as a promise.
+ */
+function getEarthquakes() {
+  var url = 'http://earthquake.usgs.gov/earthquakes/feed/v1.0/summary/all_week.geojson';
+  return Promise.resolve($.ajax({
+    dataType: 'json',
+    url: url,
+  }));
+}
+
+/**
+ * Takes a single earthquake object in as an argument, and returns a simpler
+ * earthquake containing only magnitude, age in hours, and an array of XY
+ * coordinates.
+ */
+function formatSingleEarthquake(earthquake) {
+  var earthquakeProperies = earthquake.properties;
+  var earthquakeGeometry = earthquake.geometry;
+  var earthquakeTime = earthquakeProperies.time;
+  var earthquakeAgeHours =
+    (new Date().getTime() - earthquakeTime) / (1000 * 60 * 60);
+  var earthquakeAgeHoursClean = earthquakeAgeHours.toFixed(2);
+  var earthquake = {};
+  if (earthquakeProperies.mag > 0) {
+    earthquake.magnitude = earthquakeProperies.mag;
+  } else {
+    earthquake.magnitude = .001;
+  }
+  earthquake.ageInHours = earthquakeAgeHoursClean;
+  earthquake.XYCoords = earthquakeGeometry.coordinates.slice(0, 2);
+  return earthquake;
+}
```

Comment:
Re-using the name `earthquake` for the output makes this code more confusing and requires a lot more intermediate variables.
In general, don't shadow variable names.

```js
function formatSingleEarthquake(inEarthquake) {
  return {
    ageInHours: ((new Date().getTime() - earthquakeTime) / (1000 * 60 * 60)).toFixed(2),
    xyCoords: earthquake.geometry.coordinates.slice(0, 2),
    magnitude: mag,
  };
}
```